File: HRMS/HRMS.Web/Modules/ServerTypes/HR/EmployeeDocsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DocumentKind } from "./DocumentKind";
import { EmployeeDocsRow } from "./EmployeeDocsRow";

export interface EmployeeDocsColumns {
    DocumentType: Column<EmployeeDocsRow>;
    Title: Column<EmployeeDocsRow>;
    Description: Column<EmployeeDocsRow>;
    UploadedOn: Column<EmployeeDocsRow>;
    FilePath: Column<EmployeeDocsRow>;
}

export class EmployeeDocsColumns extends ColumnsBase<EmployeeDocsRow> {
    static readonly columnsKey = 'HR.EmployeeDocs';
    static readonly Fields = fieldsProxy<EmployeeDocsColumns>();
}

[DocumentKind]; // referenced types

File: HRMS/HRMS.Web/Modules/HR/EmployeeDocs/EmployeeDocsColumns.cs
using Serenity.ComponentModel;
using System;

namespace HRMS.HR.Columns;

[ColumnsScript("HR.EmployeeDocs")]
[BasedOnRow(typeof(EmployeeDocsRow), CheckNames = true)]
public class EmployeeDocsColumns
{
    [EditLink, Width(150)]
    public DocumentKind DocumentType { get; set; }
    [EditLink, Width(200)]
    public string Title { get; set; }
    [Width(300)]
    public string Description { get; set; }
    [Width(100), AlignRight]
    public DateTime UploadedOn { get; set; }
    [Width(80)]
    public string FilePath { get; set; }
}
